examples/paho-mqtt/main.c

```diff
@@ -0,0 +1,291 @@
+{
+    enum QoS qos = QOS0;
+
+    if (argc < 3) {
+        LOG_DEBUG("usage: %s <topic name> <string msg> [QoS level]\n",
+               argv[0]);
+        return 1;
+    }
+    if (argc == 4) {
+        qos = get_qos(argv[3]);
+    }
+    MQTTMessage message;
+    message.qos = qos;
+    message.retained = IS_RETAINED_MSG;
+    message.payload = argv[2];
+    message.payloadlen = strlen(message.payload);
+
+    int rc;
+    if ((rc = MQTTPublish(&client, argv[1], &message)) < 0) {
+        LOG_ERROR("mqtt_example: Unable to publish (%d)\n", rc);
+    }
+    else {
+        LOG_INFO("mqtt_example: Message (%s) has been published to topic %s with QOS %d\n",
+                 (char *)message.payload, argv[1], (int)message.qos);
+    }
+
+    return rc;
+}
+
+static int _cmd_sub(int argc, char **argv)
+{
+    enum QoS qos = QOS0;
+
+    if (argc < 2) {
+        LOG_DEBUG("usage: %s <topic name> [QoS level]\n", argv[0]);
+        return 1;
+    }
+
+    if (argc >= 3) {
+        qos = get_qos(argv[2]);
+    }
+
+    if (topic_cnt > MAX_TOPICS) {
+        LOG_ERROR("mqtt_example: Already subscribed to max %d topics, call 'unsub' command\n", topic_cnt);
+        return -1;
+    }
+
+    if (strlen(argv[1]) > MAX_LEN_TOPIC) {
+        LOG_ERROR("mqtt_example: Not subscribing, topic too long %s\n", argv[1]);
+        return -1;
+    }
+    strncpy(_topic_to_subscribe[topic_cnt], argv[1], strlen(argv[1]));
+
+    LOG_INFO("mqtt_example: Subscribing to %s\n", _topic_to_subscribe[topic_cnt]);
+    int ret = MQTTSubscribe(&client, _topic_to_subscribe[topic_cnt], qos, _on_msg_received);
+    if (ret < 0) {
+        LOG_ERROR("mqtt_example: Unable to subscribe to %s (%d)\n", _topic_to_subscribe[topic_cnt], ret);
+        _cmd_discon(0, NULL);
+    }
+    else {
+        LOG_INFO("mqtt_example: Now subscribed to %s, QOS %d\n", argv[1], (int) qos);
+        topic_cnt++;
+    }
+    return ret;
+}
+
+static int _cmd_unsub(int argc, char **argv)
+{
+    if (argc < 2) {
+        LOG_DEBUG("usage %s <topic name>\n", argv[0]);
+        return 1;
+    }
+
+    int ret = MQTTUnsubscribe(&client, argv[1]);
+
+    if (ret < 0) {
+        LOG_ERROR("mqtt_example: Unable to unsubscribe from topic: %s\n", argv[1]);
+        _cmd_discon(0, NULL);
+    }
+    else {
+        LOG_INFO("mqtt_example: Unsubscribed from topic:%s\n", argv[1]);
+        topic_cnt--;
+    }
+    return ret;
+}
+
+static const shell_command_t shell_commands[] =
+{
+    { "con",    "connect to MQTT broker",             _cmd_con    },
+    { "discon", "disconnect from the current broker", _cmd_discon },
+    { "pub",    "publish something",                  _cmd_pub    },
+    { "sub",    "subscribe topic",                    _cmd_sub    },
+    { "unsub",  "unsubscribe from topic",             _cmd_unsub  },
+    { NULL,     NULL,                                 NULL        }
+};
+
+static unsigned char buf[BUF_SIZE];
+static unsigned char readbuf[BUF_SIZE];
```

Comment:
```suggestion
static unsigned char readbuf[BUF_SIZE];

```

Reply:
was checking the review and I don' t know what changed in this line :$

Reply:
There was an empty line added. But now it's there so that's ok.